Sources/Routing/Route.swift

```diff
@@ -0,0 +1,25 @@
+import Foundation
+import HTTP
+
+/// A representation of a `route` or endpoint in your application
+/// It is made up of the basic components of a `route` and contains
+/// optional metadata.
+public final class Route {
+    public let host: String
+    public let components: [String]
+    public let method: HTTP.Method
+    public let metadata: [String: String]?
```

Comment:
Rather than making `metadata` an optional, it could just be initialized with an empty dictionary `= [:]` by default.

I don't think it's guaranteed that the metadata will always be representable as a String. Descriptions sure can be represented as a String, but let's think about a few other use cases (such as sub-dictionaries), too